pkg/controllers/hpascaletargetmarker/hpa_scale_target_marker_controller.go

```diff
@@ -53,7 +53,7 @@ func (r *HpaScaleTargetMarker) SetupWithManager(mgr controllerruntime.Manager) e
 		ReconcileFunc: r.reconcileScaleRef,
 	}
 	r.scaleTargetWorker = util.NewAsyncWorker(scaleTargetWorkerOptions)
-	r.scaleTargetWorker.Run(scaleTargetWorkerNum, context.Background().Done())
+	r.scaleTargetWorker.Run(context.Background(), scaleTargetWorkerNum)
```

Comment:
In this case, is it better to pass ctx through the controller parameter?

Reply:
For the entire repo that's true, but I don't really want to make this change in this PR. Since this PR already contains too many modifications, it should be a pure refactor that doesn't affect any specific functionality. Introducing changes like linking the upstream and downstream `context`, which could potentially impact the code logic, would make it harder to review.

Reply:
Thanks for clarifying.